web/src/app/monitors/monitors-list/monitors-list.component.ts

```diff
@@ -0,0 +1,76 @@
+import { Component, OnInit } from '@angular/core';
+
+// Rxjs operators
+import { ActivatedRoute, Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+
+// Dashboard hub application model and services
+import { MatSnackBar } from '@angular/material';
+import { ProjectService } from '../../core/services/index.service';
+import { MonitorModel, ProjectModel } from '../../shared/models/index.model';
+
+@Component({
+  selector: 'dashboard-monitors-list',
+  templateUrl: './monitors-list.component.html',
+  styleUrls: ['./monitors-list.component.scss'],
+})
+export class MonitorsListComponent implements OnInit {
+
+  private projectSubscription: Subscription;
+  private deleteSubscription: Subscription;
+  private project: ProjectModel;
+  private uid: string;
+  public monitors: MonitorModel[] = [];
+
+  constructor(
+    private projectService: ProjectService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private snackBar: MatSnackBar
+  ) { }
+
+  /**
+   * Lifecycle init method
+   */
+  ngOnInit(): void {
+    this.uid = this.route.snapshot.paramMap.get('uid');
+    this.projectSubscription = this.projectService
+      .findOneById(this.uid)
+      .subscribe((project: ProjectModel) => {
+        this.project = project;
+        this.monitors = project.monitors ? project.monitors : [];
+      });
+  }
+
+  /**
+   * Lifecycle destroy method
+   */
+  ngDestroy(): void {
+    this.projectSubscription.unsubscribe();
+    this.deleteSubscription.unsubscribe();
+  }
+
+  /**
+   * This method is used to navigate on edit monitor screen
+   */
+  editMonitor(id: string): void {
+    this.router.navigate([`/projects/${this.uid}/monitors/${id}`]);
+  }
+
+  /**
+   * This method is used to delete the monitor from list
+   */
+  deleteMonitor(id: string): void {
+    const objIndex: number = this.monitors.findIndex((monitor: MonitorModel) => monitor.uid === id);
+    this.monitors = [
+      ...this.monitors.slice(0, objIndex),
+      ...this.monitors.slice(objIndex + 1),
```

Comment:
I think this would be cleaner with a `filter` instead of `findIndex` and `slice`

Reply:
If I will use filter then it will return me the monitor which I need for update , but I need to update the  array list with the updated monitor value so here logic is 
First I am finding the monitor which I needs to update then using slice I am updating its content.

Please let me know your thoughts on it

Reply:
With the filter you can do a not match `!` and it will return all the monitors apart from the one that got deleted.

```typescript
const monitors: MonitorModel[] = this.monitors.filter((monitor: MonitorModel) => monitor.uid !== id);
```

What do you think?

Reply:
But I have to update the one in the array

Reply:
I have 4 monitors id1, id2, id3 in an array and I have to update monitor with id2. Then how will it possible with filter ?

Reply:
Update? I think this is the delete, so just to return all the ones you want to keep and remove the one you do not?

Reply:
ops.. yes in delete this can be done. I have also used the same logic with findIndex and slice for update. So I was thinking about that. I will fix for delete. Thank you :)